doc/principles/0001-testing-principles.md

```diff
@@ -0,0 +1,34 @@
+# 1. Testing Principles
```

Comment:
Even though these are principles, they still do define the architecture of our code base. In that sense, maybe they should live alongside ADRs? (`doc/architecture/decisions`).